test/test.js

```diff
@@ -1346,6 +1325,37 @@ The prop passed to useFragment() should be typed with the type 'TestFragment_foo
           column: 15
         }
       ]
+    },
+    {
+      filename: 'MyComponent.jsx',
+      code: `
+        type RelayProps = {
+          user: MyComponent_user
+        }
+
+        type Props = {
+          other: ?Object,
+        } & RelayProps;
+
+        class MyComponent extends React.Component<Props> {
+          render() {
+            return <div />;
+          }
+        }
+
+        createFragmentContainer(MyComponent, {
+          user: graphql\`fragment MyComponent_user on User {id}\`,
+        });
+      `,
+      errors: [
+        {
+          message:
+            '`user` is not declared in the `props` of the React component or it is not marked with the generated flow type `MyComponent_user`. ' +
```

Comment:
Hi @josephsavona! Could you please check this issue? https://github.com/relayjs/eslint-plugin-relay/issues/59 I think it's related to what you wrote here since seems like it doesn't work as expected (even though in this case it's probably fine).

Reply:
That looks like a separate issue. The issue here is just that we’re not following type aliases and not unwrapping readonly. Happy to accept PRs for #59 though!